eth/stagedsync/stage_custom_trace.go

```diff
@@ -125,17 +125,22 @@ func SpawnCustomTrace(cfg CustomTraceCfg, ctx context.Context, logger log.Logger
 	return nil
 }
 
-func customTraceBatchProduce(ctx context.Context, cfg *exec3.ExecArgs, db kv.RwDB, fromBlock, toBlock uint64, logPrefix string, logger log.Logger) error {
+func customTraceBatchProduce(ctx context.Context, cfg *exec3.ExecArgs, db kv.TemporalRwDB, fromBlock, toBlock uint64, logPrefix string, logger log.Logger) error {
 	var lastTxNum uint64
-	if err := db.Update(ctx, func(tx kv.RwTx) error {
-		ttx := tx.(kv.TemporalRwTx)
+	{
+		tx, err := db.BeginTemporalRw(ctx)
+		defer tx.Rollback()
```

Comment:
this should be after err check

Reply:
FYI: we have linter for this https://github.com/erigontech/erigon/actions/runs/13585116850/job/37978304733?pr=14009#step:4:26
they are in `rules.go`